runtime/startup_aux.c

```diff
@@ -186,12 +190,21 @@ CAMLexport void caml_shutdown(void)
 
   call_registered_value("Pervasives.do_at_exit");
   call_registered_value("Thread.at_shutdown");
-  caml_finalise_heap();
+  if (!caml_domain_alone()) {
+    caml_gc_log("Some domains have not been joined prior to shutdown");
+    caml_stop_all_domains();
```

Comment:
We try to stop all domains not just when the memory cleanup at exit mode is on. Have we decided that this is what we want?

Reply:
This is `caml_shutdown`, which is only invoked in cleanup-at-exit mode, see `runtime/sys.c` lines 193-194.